osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,764 @@
+                         Row& r) {
+  if ((stream == nullptr) || (params == nullptr))
+    return Status(1);
+
+  // Offset 0x80 is from WinDbg: dt nt!_RTL_USER_PROCESS_PARAMETERS
+  auto envVarsAddr = reinterpret_cast<const unsigned long long*>(
+      reinterpret_cast<const unsigned char*>(params) + 0x80);
+  auto result = getMemAtTarget(stream, *envVarsAddr, dumpBase);
+  if (result == nullptr)
+    return Status(1);
+  auto envVars = static_cast<wchar_t*>(result);
+
+  // Loop through environment variables and log those of interest
+  // The environment variables are stored in the following format:
+  // Var1=Value1\0Var2=Value2\0Var3=Value3\0 ... VarN=ValueN\0\0
+  wchar_t* ptr = envVars;
+  while (*ptr != '\0') {
+    auto envVar = wstringToString(std::wstring(ptr).c_str());
+    auto pos = envVar.find('=');
+    auto varName = envVar.substr(0, pos);
+    auto varValue = envVar.substr(pos + 1, envVar.length());
+
+    if (varName == "COMPUTERNAME") {
+      r["machine_name"] = varValue;
+    } else if (varName == "USERNAME") {
+      r["username"] = varValue;
+    }
+
+    ptr += envVar.length() + 1;
+  }
+  return Status(0);
+}
+
+Status logDumpTime(const MINIDUMP_HEADER* header, Row& r) {
+  if (header == nullptr)
+    return Status(1);
+
+  time_t dumpTimestamp = header->TimeDateStamp;
+  struct tm gmt;
+  char timeBuff[64];
+  gmtime_s(&gmt, &dumpTimestamp);
+  strftime(timeBuff, sizeof(timeBuff), "%Y-%m-%d %H:%M:%S UTC", &gmt);
+  r["datetime"] = timeBuff;
+  return Status(0);
+}
+
+Status logAndStoreDumpType(const MINIDUMP_HEADER* header,
+                           unsigned long long& flags,
+                           Row& r) {
+  if (header == nullptr)
+    return Status(1);
+
+  std::ostringstream activeFlags;
+  bool firstString = true;
+  // Loop through MINIDUMP_TYPE flags and log the ones that are set
+  for (auto const& flag : kMinidumpTypeFlags) {
+    if (header->Flags & flag.first) {
+      if (!firstString)
+        activeFlags << ",";
+      firstString = false;
+      activeFlags << flag.second;
+    }
+  }
+  r["type"] = activeFlags.str();
+  flags = header->Flags;
+  return Status(0);
+}
+
+void debugEngineCleanup(IDebugClient4* client,
+                        IDebugControl4* control,
+                        IDebugSymbols3* symbols) {
+  if (symbols != nullptr)
+    symbols->Release();
+  if (control != nullptr)
+    control->Release();
+  if (client != nullptr) {
+    client->SetOutputCallbacks(NULL);
+    client->EndSession(DEBUG_END_PASSIVE);
+    client->Release();
+  }
+  return;
+}
+
+/*
+Note: appears to only detect unmanaged stack frames.
+See http://blog.steveniemitz.com/building-a-mixed-mode-stack-walker-part-2/
+*/
+void logStackTrace(const char* fileName, Row& r) {
+  IDebugClient4* client;
+  IDebugControl4* control;
+  IDebugSymbols3* symbols;
+  DEBUG_STACK_FRAME stackFrames[kNumStackFramesToLog] = {0};
+  unsigned long numFrames = 0;
+  char context[1024] = {0};
+  unsigned long type = 0;
```

Comment:
For each of these local variables, and try to use this same logic throughout your code base, try to only initialize the variable immediately before it's used. Most of these variables aren't used for quite a few lines of code, so bring them further down to just above their use case.